portfolio/src/main/webapp/calendar.js

```diff
@@ -69,14 +103,70 @@ function getTimeObject(timeString) {
   return new Date(currentYear, currentMonth, currentDate, timeHour, timeMinute);
 }
 
-/** Creates a single element for a new calendar event. */
+/** Creates a card element for a new calendar event. */
+/** 
+ * TODO(hollyyuqizheng) Look into refactoring the card creation code 
+ * into a JavaScript class. This card element will be used in different 
+ * sections on the UI, so this refactoring will make things simpler. 
+ */
 function updateCalendarEventList(newCalendarEvent) {
```

Comment:
(Optional) You can potentially have the template element in the DOM. It can have the appropriate class names and the nested structure. Maybe it can have the d-none class name (bootstrap) to be hidden. Then you can use jquery clone method to clone it here.
https://www.w3schools.com/jquery/html_clone.asp